lib/Libssh/Session.pm

```diff
@@ -430,6 +430,73 @@ sub auth_none {
     return $ret;
 }
 
+sub auth_kbdint {
+    my ($self, %options) = @_;
+
+    my $ret = ssh_userauth_kbdint($self->{ssh_session});
+    if ($ret == SSH_AUTH_ERROR) {
+        $self->set_err(msg => sprintf("authentification failed: %s", ssh_get_error_from_session($self->{ssh_session})));
+    }
+    $self->{authenticated} = 1 if ($ret == SSH_OK);
+
+    return $ret;
+}
+
+sub auth_kbdint_getnprmopts {
+    my ($self, %options) = @_;
+
+    my $ret = ssh_userauth_kbdint_getnprompts($self->{ssh_session});
+    if ($ret == SSH_AUTH_ERROR) {
```

Comment:
I don't see that the function can return SSH_AUTH_ERROR:
http://api.libssh.org/master/group__libssh__auth.html#gacb996ff4979670db009a71a90172ece9

Reply:
Looking at the source code from the 'libssh' repo it returns SSH_ERROR in some cases. You're right that I should probably replace SSH_AUTH_ERROR with that instead:

`int ssh_userauth_kbdint_getnprompts(ssh_session session) {
    if (session == NULL) {
        return SSH_ERROR;
    }
    if (session->kbdint == NULL) {
        ssh_set_error_invalid(session);
        return SSH_ERROR;
    }
    return session->kbdint->nprompts;
}`